### PR TITLE
[INLONG-6998][Dashboard][Manager] Support ignore deserialization error

### LINKED ISSUE #6998: [Feature][Dashboard][Manager] Support ignore deserialization error data
### Description

image

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by featzhang**
### Prepare a Pull Request
*[INLONG-6998][Dashboard][Manager] Support ignore deserialization erro…*

- Fixes #6998 

### Motivation

When the reported message contains garbage data, any problem of the message format will cause the Flink job to automatically restart until the data in the MQ is repaired; 
Otherwise, all the data is piled into the MQ to continue processing.

This option `ignoreParseError` is exposed in the hope that the user can adjust it according to the data format requirements.

### Modifications

Pass the attribute `ignoreParseError` to the concrete implementation of format, which is enabled by default.

image
image

The follow sql generated:

```sql
CREATE TABLE `table_s-v3`(
    `uid` INT,
    `uname` STRING,
    `cata` STRING,
    `eid` STRING,
    `fee` DOUBLE,
    `ftime` TIMESTAMP(2))
    WITH (
    'inlong.metric.labels' = 'groupId=sg-v2&streamId=s-v3&nodeId=s-v3',
    'metrics.audit.proxy.hosts' = '127.0.0.1:10081',
    'connector' = 'pulsar-inlong',
    'csv.allow-comments' = 'false',
    'csv.disable-quote-character' = 'true',
    'csv.array-element-delimiter' = ';',
    'csv.ignore-parse-errors' = 'false',
    'format' = 'csv',
    'csv.field-delimiter' = ',',
    'admin-url' = 'http://127.0.0.1:8080',
    'generic' = 'true',
    'service-url' = 'pulsar://127.0.0.1:6650',
    'topic' = 'public/sg-v2/s-v3',
    'scan.startup.mode' = 'earliest'
) 
```

The extended params `ignoreParseError` and `wrapWithInlongMsg` will saved as `ExtList` in `inlong_stream_ext` table of database:

image

### Verifying this change

*(Please pick either of the following options)*

- [x] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Comment by dockerzhang** [comment]
Please add more details about why we could ignore the deserialization error

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/tubemq/TubeMQSourceOperator.java [hunk 1]
```diff
@@ -107,6 +107,8 @@ public Map<String, List<StreamSource>> getSourcesMap(InlongGroupInfo groupInfo,
             tubeMQSource.setTopic(streamInfo.getMqResource());
             tubeMQSource.setGroupId(streamId);
             tubeMQSource.setMasterRpc(masterRpc);
+            tubeMQSource.setSerializationType(streamInfo.getDataType());
+
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/tubemq/TubeMQSourceOperator.java:111]: Why TubeMQ source not add setIgnoreFormatParseError?

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/source/StreamSource.java [hunk 2]
```diff
@@ -110,6 +110,9 @@ public abstract class StreamSource extends StreamNode {
     @ApiModelProperty("Sub source information of existing agents")
     private List<SubSourceDTO> subSourceList;
 
+    @ApiModelProperty(value = "Whether ignore deserialization error, true as default")
+    private boolean ignoreFormatParseError;
```
Comment (healchow) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/source/StreamSource.java:114]:
```suggestion

ignoreParseError
    @ApiModelProperty(value = "Whether to ignore the parse errors of field value, true as default")
    private boolean ignoreParseError;
```

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/stream/InlongStreamInfo.java [hunk 3]
```diff
@@ -42,6 +42,7 @@
 @ApiModel("Inlong stream info")
 public class InlongStreamInfo extends BaseInlongStream {
 
+    private static final int DEFAULT_ENABLE_VALUE = 1;
```
Comment (healchow) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/stream/InlongStreamInfo.java:45]: Suggest moving to `InlongConstants`.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/stream/InlongStreamInfo.java [hunk 4]
```diff
@@ -127,6 +128,13 @@ public class InlongStreamInfo extends BaseInlongStream {
     @ApiModelProperty(value = "Version number")
     private Integer version;
 
+    @ApiModelProperty(value = "Whether ignore deserialization error, 0: no, 1: yes (as default)")
+    private Integer formatIgnoreParseError;
```
Comment (healchow) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/stream/InlongStreamInfo.java:132]:
```suggestion
    @ApiModelProperty(value = "Whether to ignore the parse errors of field value, 0: no, 1: yes (as default)")
    private Integer ignoreParseError;
```

**Review by healchow — CHANGES_REQUESTED** [review]

File: inlong-dashboard/src/locales/en.json [hunk 5]
```diff
@@ -367,6 +367,7 @@
   "meta.Stream.DataEncoding": "Data encoding",
   "meta.Stream.Description": "Description",
   "meta.Stream.SourceDataField": "Source fields",
+  "meta.Stream.FormatIgnoreParseError": "ignoreParseError",
```
Comment (healchow) [inline inlong-dashboard/src/locales/en.json:370]:
```suggestion
  "meta.Stream.IgnoreParseError": "Ignore parse error",
```

File: inlong-dashboard/src/locales/cn.json [hunk 6]
```diff
@@ -367,6 +367,7 @@
   "meta.Stream.DataEncoding": "数据编码",
   "meta.Stream.Description": "介绍",
   "meta.Stream.SourceDataField": "源数据字段",
+  "meta.Stream.FormatIgnoreParseError": "忽略字段解析异常",
```
Comment (healchow) [inline inlong-dashboard/src/locales/cn.json:370]:
```suggestion
  "meta.Stream.IgnoreParseError": "忽略数据解析错误",
```

**Review by healchow — APPROVED** [review]
LGTM

**Review by leezng — APPROVED** [review]

**Review by fuweng11 — APPROVED** [review]

**Comment by dockerzhang** [comment]
@featzhang please rebase from the master and fix the conflict.

**Review by woofyzhao — APPROVED** [review]

**Comment by dockerzhang** [comment]
@healchow @fuweng11 PTAL again, thanks.

**Comment by dockerzhang** [comment]
@featzhang could we move the parse error to the advanced options?

**Comment by featzhang** [comment]
> @featzhang could we move the parse error to the advanced options?

It involves the transformation of the front-end module. The larger action will be processed in another pr. I hope the front-end students will support it.

**Review by yunqingmoswu — APPROVED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/InlongStreamServiceImpl.java [hunk 7]
```diff
@@ -514,12 +571,35 @@ public Boolean updateWithoutCheck(InlongStreamRequest request, String operator)
...
         return true;
     }
 
+    /**
+     * Pack extended attributes into {@link InlongStreamExtInfo} 
+     * @param request the request
+     * @param extList the extList from frontend
+     * @return the packed extParams
+     */
+    private List<InlongStreamExtInfo> packExtParams(InlongStreamRequest request,
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/InlongStreamServiceImpl.java:588]: I don't think additional parameters can be added in this way.If there are more parameters later, the code will become very redundant.

Comment (featzhang) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/InlongStreamServiceImpl.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/InlongStreamServiceImpl.java [hunk 8]
```diff
@@ -514,12 +569,31 @@ public Boolean updateWithoutCheck(InlongStreamRequest request, String operator)
...
     }
 
+    /**
+     * Pack extended attributes into ExtParams 
+     *
+     * @param request the request
+     * @return the packed extParams
+     */
+    private String packExtParams(InlongStreamRequest request) {
+        InlongStreamExtParamBuilder builder = InlongStreamExtParam.builder();
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/InlongStreamServiceImpl.java:586]: It is recommended to use `CommonBeanUtils.copyProperties(request, InlongStreamExtParam::new, true);`

Comment (featzhang) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/InlongStreamServiceImpl.java:586]: The `org.apache.inlong.manager.dao.entity.InlongStreamEntity`
does not contain the same fields, json must be compressed into extParams, so `CommonBeanUtils.copyProperties(request, InlongStreamExtParam::new, true)` cannot be used;

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/InlongStreamServiceImpl.java [hunk 9]
```diff
@@ -385,6 +424,16 @@ public PageResult<InlongStreamInfo> listAll(InlongStreamPageRequest request) {
         return pageResult;
     }
 
+    private void unpackExtParams(InlongStreamInfo streamInfo) {
+        String extParams = streamInfo.getExtParams();
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/InlongStreamServiceImpl.java:428]: It seems that there is a method with the same function in line 244. It is recommended to unify the two methods

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/InlongStreamServiceImpl.java (hunk 8 above)
Comment (featzhang) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/InlongStreamServiceImpl.java]: ok, done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/InlongStreamServiceImpl.java (hunk 9 above)
Comment (featzhang) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/InlongStreamServiceImpl.java]: ok, done

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/stream/InlongStreamExtParam.java [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+/**
+ * Extended params, will be saved as JSON string
+ */
+@Data
+@EqualsAndHashCode(callSuper = false)
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@ApiModel("Inlong stream info")
```
Comment (fuweng11) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/stream/InlongStreamExtParam.java:37]: Inlong stream ext param info.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/kafka/KafkaSourceOperator.java [hunk 11]
```diff
@@ -111,16 +110,19 @@ public Map<String, List<StreamSource>> getSourcesMap(InlongGroupInfo groupInfo,
             kafkaSource.setSourceName(streamId);
             kafkaSource.setBootstrapServers(bootstrapServers);
             kafkaSource.setTopic(streamInfo.getMqResource());
+            kafkaSource.setSerializationType(streamInfo.getDataType());
+            String serializationType = DataTypeEnum.forType(streamInfo.getDataType()).getType();
+            kafkaSource.setSerializationType(serializationType);
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/kafka/KafkaSourceOperator.java:115]: Why set the value twice?

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/InlongStreamServiceImpl.java [hunk 12]
```diff
@@ -514,12 +562,25 @@ public Boolean updateWithoutCheck(InlongStreamRequest request, String operator)
...
         return true;
     }
 
+    /**
+     * Pack extended attributes into ExtParams 
+     *
+     * @param request the request
+     * @return the packed extParams
+     */
+    private String packExtParams(InlongStreamRequest request) {
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/InlongStreamServiceImpl.java:578]: It is recommended to put it in the class `inlongStreamExtParam`.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/InlongStreamServiceImpl.java [hunk 13]
```diff
@@ -226,6 +240,27 @@ public InlongStreamInfo get(String groupId, String streamId) {
         return streamInfo;
     }
 
+    /**
+     * Unpack extended attributes from {@link InlongStreamExtInfo}, will remove target attributes from it.
+     *
+     * @param extParams the {@link InlongStreamEntity} load from db 
+     * @param targetObject the targetObject with to fill up
+     */
+    private void unpackExtParams(
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/InlongStreamServiceImpl.java:249]: It is recommended to put it in the class `inlongStreamExtParam`.

**Review by healchow — APPROVED** [review]

**Pull request merged**